src/common/dao/blob.go

```diff
@@ -50,15 +64,73 @@ func DeleteBlob(digest string) error {
 	return err
 }
 
-// HasBlobInProject ...
-func HasBlobInProject(projectID int64, digest string) (bool, error) {
-	var res []orm.Params
-	num, err := GetOrmer().Raw(`SELECT * FROM artifact af LEFT JOIN artifact_blob afnb ON af.digest = afnb.digest_af WHERE af.project_id = ? and afnb.digest_blob = ? `, projectID, digest).Values(&res)
+// GetBlobsByArtifact returns blobs of artifact
+func GetBlobsByArtifact(artifactDigest string) ([]*models.Blob, error) {
+	sql := `SELECT * FROM blob WHERE digest IN (SELECT digest_blob FROM artifact_blob WHERE digest_af = ?)`
+
+	var blobs []*models.Blob
+	if _, err := GetOrmer().Raw(sql, artifactDigest).QueryRows(&blobs); err != nil {
+		return nil, err
+	}
+
+	return blobs, nil
+}
+
+// GetExclusiveBlobs returns layers of repository:tag which are not shared with other repositories in the project
+func GetExclusiveBlobs(projectID int64, repository, digest string) ([]*models.Blob, error) {
+	blobs, err := GetBlobsByArtifact(digest)
```

Comment:
This returns the blobs of artifact which even isn't in the project specified by the projectID, is this the expected behavior?